pl_examples/basic_examples/mnist_examples/image_classifier_3_lite_to_lightning.py

```diff
@@ -0,0 +1,168 @@
+
+        train_loader, test_loader = self.setup_dataloaders(self.train_dataloader(), self.train_dataloader())
+        self.test_acc = Accuracy()
+
+        # EPOCH LOOP
+        for epoch in range(1, hparams.epochs + 1):
+
+            # TRAINING LOOP
+            self.model.train()
+            for batch_idx, batch in enumerate(train_loader):
+                optimizer.zero_grad()
+                loss = self.training_step(batch, batch_idx)
+                self.backward(loss)
+                optimizer.step()
+
+                if (batch_idx == 0) or ((batch_idx + 1) % hparams.log_interval == 0):
+                    print(
+                        "Train Epoch: {} [{}/{} ({:.0f}%)]\tLoss: {:.6f}".format(
+                            epoch,
+                            (batch_idx + 1) * self.hparams.batch_size,
+                            len(train_loader.dataset),
+                            100.0 * batch_idx / len(train_loader),
+                            loss.item(),
+                        )
+                    )
+                    if hparams.dry_run:
+                        break
+
+            scheduler.step()
+
+            # TESTING LOOP
+            self.model.eval()
+            test_loss = 0
+            with torch.no_grad():
+                for batch_idx, batch in enumerate(test_loader):
+                    test_loss += self.test_step(batch, batch_idx)
+                    if hparams.dry_run:
+                        break
+
+            test_loss = self.all_gather(test_loss).sum() / len(test_loader.dataset)
+
+            print(f"\nTest set: Average loss: {test_loss:.4f}, Accuracy: ({self.test_acc.compute():.0f}%)\n")
+            self.test_acc.reset()
+
+            if hparams.dry_run:
+                break
+
+        if hparams.save_model:
+            self.save(model.state_dict(), "mnist_cnn.pt")
+
+    # Methods for the `LightningModule` conversion
+
+    def forward(self, x):
+        return self.model(x)
+
+    def training_step(self, batch, batch_idx):
+        """Here you compute and return the training loss and compute extra training metrics."""
+        x, y = batch
+        logits = self.forward(x)
+        loss = F.nll_loss(logits, y.long())
+        return loss
+
+    def test_step(self, batch, batch_idx):
+        """Here you compute and return the testing loss and compute extra testing metrics."""
+        x, y = batch
+        logits = self.forward(x)
+        loss = F.nll_loss(logits, y.long())
+        self.test_acc(logits, y.long())
+        return loss
+
+    def configure_optimizers(self):
+        optimizer = optim.Adadelta(self.model.parameters(), lr=self.hparams.lr)
+        return [optimizer], [StepLR(optimizer, step_size=1, gamma=self.hparams.gamma)]
+
+    # Methods for the `LightningDataModule` conversion
+
+    @property
+    def transform(self):
+        return T.Compose([T.ToTensor(), T.Normalize((0.1307,), (0.3081,))])
+
+    def prepare_data(self) -> None:
```

Comment:
```suggestion
    @staticmethod
    def prepare_data() -> None:
```

Reply:
prepare_data on the datamodule is also not a static method
it would be better to stay consistent, otherwise it is confusing to the user